### PR TITLE
Debugged Properties panel close Animation Bug

### PR DESCRIPTION
**Pull request by PrakharJain1509**
# Fixing Overflow Issue in EditPanel Component

## Problem
While opening or closing the `EditPanel` tab, the content inside was overflowing beyond the boundaries of the panel. This caused the UI to break during animations.

## Solution
We applied **`overflow-hidden`** to critical containers and added **width/height constraints** to prevent content from escaping during transitions. We also ensured smooth transitions with CSS adjustments.

### Changes Made

1. **Apply `overflow-hidden` to the Main Wrapper:**
   - Ensures that the content stays within the panel boundaries during open/close transitions.
   ```jsx
   <div
       className={clsx(
           'fixed right-0 transition-width duration-300 opacity-100 bg-background/80 rounded-tl-xl overflow-hidden',
           editorEngine.mode === EditorMode.INTERACT ? 'hidden' : 'visible',
           !isOpen && 'w-12 h-12 rounded-l-xl cursor-pointer',
           isOpen && 'h-[calc(100vh-5rem)]',
           isOpen && selectedTab == TabValue.STYLES && 'w-60',
           isOpen && selectedTab == TabValue.CHAT && 'w-[22rem]',
       )}
   >

### CODE REVIEW

**Comment by PrakharJain1509** [comment]
@drfarrell @Kitenite here is the PR for that same, This has successfully debugged that issue and the text is now not coming out of the transition box. Please review it and let me know if any changes are required.

**Comment by PrakharJain1509** [comment]
This was for #651 .

**Review by Kitenite — CHANGES_REQUESTED** [review]
This is certainly better! However, I'm seeing some regression from this change:

1. the tailwind class box is larger than it should be. 
2. The editor panel no longer scrolls

Screenshot 2024-10-25 at 4 16 34 PM

**Comment by PrakharJain1509** [comment]
@Kitenite Please check now, Scroll is working now, and the tailwind class is also in size.

**Comment by PrakharJain1509** [comment]
@Kitenite Any updates sir?

File: app/src/routes/editor/EditPanel/StylesTab/index.tsx [hunk 1]
```diff
@@ -104,7 +104,7 @@ const ManualTab = observer(() => {
                 <AccordionTrigger>
                     <h2 className="text-xs font-semibold">{groupKey}</h2>
                 </AccordionTrigger>
-                <AccordionContent>
+                <AccordionContent className="overflow-auto max-h-[50vh]">
```
Comment (Kitenite) [inline app/src/routes/editor/EditPanel/StylesTab/index.tsx:107]: Can you explain why this is necessary? Seems like we just need `overflow-hidden` on the root panel div?

File: app/src/routes/editor/EditPanel/StylesTab/index.tsx [hunk 2]
```diff
@@ -128,7 +128,7 @@ const ManualTab = observer(() => {
     return (
         editorEngine.elements.selected.length > 0 && (
             <Accordion
-                className="px-4"
+                className="px-4 overflow-hidden"
```
Comment (Kitenite) [inline app/src/routes/editor/EditPanel/StylesTab/index.tsx:131]: Same with this

File: app/src/routes/editor/EditPanel/index.tsx [hunk 3]
```diff
@@ -77,38 +77,38 @@ const EditPanel = observer(() => {
...
+                                <Icons.CrossS/>
                             </Button>
                         </div>
                     )}
                 </TabsList>
-                <Separator />
+                <Separator/>
                 <div className="h-[calc(100vh-7.75rem)] overflow-auto">
-                    <TabsContent value={TabValue.STYLES}>
+                    <TabsContent value={TabValue.STYLES} className="min-w-0 max-w-full min-h-0 max-h-full">
```
Comment (Kitenite) [inline app/src/routes/editor/EditPanel/index.tsx:102]: And these 2

**Review by Kitenite — COMMENTED** [review]
Just one question, seems like there are more overflow-hidden implemented in code than what was suggested in the description. Is this correct? 
Also should we do the same thing in the LayersPanel?

**Comment by Kitenite** [comment]
> @Kitenite Any updates sir?

Sorry I was working on a different PR. Left some comments but thanks for the quick turnaround!

File: app/src/routes/editor/EditPanel/StylesTab/index.tsx (hunk 1 above)
Comment (PrakharJain1509) [reply app/src/routes/editor/EditPanel/StylesTab/index.tsx:107]: It creates issues with the Scroll if I do just that.

**Comment by PrakharJain1509** [comment]
> > @Kitenite Any updates sir?
> 
> Sorry I was working on a different PR. Left some comments but thanks for the quick turnaround!

So I was not sure where the code was at first so was experimenting, So is it fine now ?

**Comment by PrakharJain1509** [comment]
@Kitenite Did it for the layers too.

**Review by Kitenite — APPROVED** [review]
Working well! Thanks for being very responsive on this

**Comment by PrakharJain1509** [comment]
@Kitenite Can you please add the hacktoberfest-accepted label also, for it to get counted.

**Comment by Kitenite** [comment]
Ofc!

**Pull request merged**